src/main/java/org/aigents/nlp/gen/Segment.java

```diff
@@ -99,17 +118,51 @@ private static boolean check(String[] input) {
 			return false;
 		if (last.equals("on") || first.equals("on"))
 			return false;
-		if (last.equals("with") || first.equals("with"))
-			return false;
-		if (first.equals("board"))
+		if (last.equals("with") || last.equals("the") || first.equals("with"))
 			return false;
-		if (first.equals("in") || last.equals("in")) return false;
-		if (first.equals("writes") || first.equals("wants") || first.equals("has") || first.equals("sees")
-				|| first.equals("sawed") || first.equals("knocked") || first.startsWith("like") || first.equals("is")
-				|| first.equals("was")) {
+		if (first.equals("board") || last.equals("been") || last.equals("at")) return false;
```

Comment:
Seems like we are hardcoding rules very specific to particular testing corpus which seem even less fair than hardcoding the grammar... Why do we need to do that?